### PR TITLE
Support writing TIMESTAMPTZ with JSON encoder

### LINKED ISSUE #5955: Extend support for INSERT TIME/TIMESTAMP WITH TIMEZONE in Kafka connector JSON encoder
Allow INSERT of temporal types with time-zone in Kafka connector with JSON Encoder when using the `milliseconds-since-epoch` or `seconds-since-epoch` formatters.

See discussion at https://github.com/prestosql/presto/pull/5838#discussion_r520477341.

### PR DESCRIPTION
**Pull request by hashhar**
Fixes #5955.

### CODE REVIEW

**Review by findepi — COMMENTED** [review]
LGTM, can you please add a test and make it non-draft?

**Comment by hashhar** [comment]
@findepi I added some tests but really think there should be a better way of generating the expected values. The current one is error-prone (maybe even wrong).

Tests for seconds-since-epoch and more test values I'll push in a while.

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestSecondsJsonDateTimeFormatter.java [hunk 1]
```diff
@@ -57,5 +78,12 @@ public void testSecondsDateTimeFunctions()
         testTimestamp(sqlTimestampOf(3, 2020, 8, 18, 12, 38, 29, 0), LocalDateTime.of(2020, 8, 18, 12, 38, 29, 0).toEpochSecond(ZoneOffset.UTC));
         testTimestamp(sqlTimestampOf(3, 1970, 1, 1, 0, 0, 0, 0), 0);
         testTimestamp(sqlTimestampOf(3, 1800, 8, 18, 12, 38, 29, 0), LocalDateTime.of(1800, 8, 18, 12, 38, 29, 0).toEpochSecond(ZoneOffset.UTC));
+
+        testTimestampWithTZ(sqlTimestampWithTimeZoneOf(3, 2020, 8, 18, 12, 38, 29, 123000000, UTC_KEY), getEpochSecondsFromDateTimeWithZone(2020, 8, 18, 12, 38, 29, 123000000, UTC_KEY));
```
Comment (findepi) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestSecondsJsonDateTimeFormatter.java:82]: that's very repetitive, but maybe that's fine... let's format this on lines:

```
        testTimestampWithTZ(
                sqlTimestampWithTimeZoneOf(3, 2020, 8, 18, 12, 38, 29, 123000000, UTC_KEY),
                ZonedDateTime.of(2020, 8, 18, 12, 38, 29, 123000000, UTC).toEpochSecond());
```

and remove `getEpochSecondsFromDateTimeWithZone` method

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestMillisecondsJsonDateTimeFormatter.java [hunk 2]
```diff
@@ -49,6 +56,12 @@ private void testTimestamp(SqlTimestamp value, long actualMillis)
         assertEquals(Long.parseLong(formattedStr), actualMillis);
     }
 
+    private void testTimestampWithTZ(SqlTimestampWithTimeZone value, long actualMillis)
+    {
+        String formattedStr = getFormatter().formatTimestampWithZone(value);
```
Comment (findepi) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestMillisecondsJsonDateTimeFormatter.java:61]: formattedStr -> formatted

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestMillisecondsJsonDateTimeFormatter.java [hunk 3]
```diff
@@ -49,6 +56,12 @@ private void testTimestamp(SqlTimestamp value, long actualMillis)
         assertEquals(Long.parseLong(formattedStr), actualMillis);
     }
 
+    private void testTimestampWithTZ(SqlTimestampWithTimeZone value, long actualMillis)
```
Comment (findepi) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestMillisecondsJsonDateTimeFormatter.java:59]: actualMillis -> expected

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestSecondsJsonDateTimeFormatter.java [hunk 4]
```diff
@@ -35,27 +38,57 @@ private JsonDateTimeFormatter getFormatter()
...
+                {LocalDateTime.of(1970, 1, 1, 0, 0, 0, 0)},
+                {LocalDateTime.of(1800, 8, 18, 12, 38, 29, 0)},
+        };
+    }
 
-        testTimestamp(sqlTimestampOf(3, 2020, 8, 18, 12, 38, 29, 0), LocalDateTime.of(2020, 8, 18, 12, 38, 29, 0).toEpochSecond(ZoneOffset.UTC));
-        testTimestamp(sqlTimestampOf(3, 1970, 1, 1, 0, 0, 0, 0), 0);
-        testTimestamp(sqlTimestampOf(3, 1800, 8, 18, 12, 38, 29, 0), LocalDateTime.of(1800, 8, 18, 12, 38, 29, 0).toEpochSecond(ZoneOffset.UTC));
+    @Test(dataProvider = "testTimestampWithTimeZoneProvider")
+    public void testTimestampWithTimeZone(ZonedDateTime zonedDateTime)
```
Comment (findepi) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestSecondsJsonDateTimeFormatter.java:76]: can you please extract test splitting as a prep commit?

(i am aware you would need to squash&rebase the existing work piece for that)

Comment (hashhar) [reply presto-kafka/src/test/java/io/prestosql/plugin/kafka/encoder/json/TestSecondsJsonDateTimeFormatter.java:76]: Done. I'll do a follow up to clean-up some other test classes in the same module.

**Review by findepi — APPROVED** [review]

**Pull request merged**